stackslib/src/chainstate/stacks/block.rs

```diff
@@ -567,38 +577,86 @@ impl StacksBlock {
     pub fn validate_transactions_static_epoch(
         txs: &[StacksTransaction],
         epoch_id: StacksEpochId,
+        quiet: bool,
     ) -> bool {
         for tx in txs.iter() {
             if let TransactionPayload::Coinbase(_, ref recipient_opt, ref proof_opt) = &tx.payload {
                 if proof_opt.is_some() && epoch_id < StacksEpochId::Epoch30 {
                     // not supported
-                    error!("Coinbase with VRF proof not supported before Stacks 3.0"; "txid" => %tx.txid());
+                    if !quiet {
```

Comment:
Maybe use a macro to avoid all these `if` statements:

```rust
// Quietable error
macro_rules! qerror {
    ($cond: expr, $($arg:tt)*) => ({
        if ($cond) {
            error!($($arg)*);
        }
    });
}
```

```rust
qerror!(quiet, "Coinbase with VRF proof is required in Stacks 3.0 and later"; "txid" => %tx.txid());
```